rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala

```diff
@@ -730,38 +711,36 @@ class InternalLDAPQueryProcessor(
    */
   private def normalize(query:Query) : Box[LDAPNodeQuery] = {
 
-    //validate that we knows the requested object type
-    if(!objectTypes.isDefinedAt(query.returnType.value))
-      return Failure("The requested type '%s' is not know".format(query.returnType))
-
-    //validate that for each object type in criteria, we know it
-    query.criteria foreach { cl =>
-      if(!objectTypes.isDefinedAt(cl.objectType.objectType))
-        return Failure("The object type '%s' is not know in criteria %s".format(cl.objectType.objectType,cl))
+    def groupFilterByLdapRequest() = {
+      //validate that we knows the requested object type
+      if (objectTypes.isDefinedAt(query.returnType.value)) {
+        (sequence(query.criteria) {
+           case crit@CriterionLine(ot,a,comp,value) => {
+            val objectType = ot.objectType
```

Comment:
i'm not 100% sure, but it seems the semantic changed compared to what was defined before:
* we had criterai grouped by objectType, and Filter generated for each group of object class
* now for each criteria we create the filter, map the object type and group by dnObjectType , and then two mapValues (which are super hard to comprehend) that group by resulting LDAPObjectType and then by filter
So it seems we have an extra reordering.